test/sqllogictest/object_ownership.slt

```diff
@@ -2388,13 +2388,13 @@ SELECT mz_sources.name, mz_roles.name
   WHERE mz_sources.name IN ('s', 's_progress', 'accounts', 'auctions', 'bids', 'organizations', 'users')
   ORDER BY mz_sources.name
 ----
-accounts       joe
-auctions       joe
-bids           joe
-organizations  joe
+accounts       materialize
+auctions       materialize
+bids           materialize
+organizations  materialize
```

Comment:
+1 I'm pretty sure this shouldn't change

Reply:
I believe the updated test in my commit reflects the behavior we expect. We can see that changing an owner does not change the ownership of the object's dependents using this example.

Log in as `mz_system`

```
psql -h localhost -U mz_system materialize -p 6877
```

Then:

```
SET CLUSTER TO default;
CREATE TABLE t (a int);
CREATE MATERIALIZED VIEW v AS SELECT sum(a) FROM t;
CREATE ROLE joe;

SELECT mz_tables.name AS table_name, mz_roles.name AS role_name
  FROM mz_tables
  LEFT JOIN mz_roles ON mz_tables.owner_id = mz_roles.id
  WHERE mz_tables.name = 't';

 table_name | role_name 
------------+-----------
 t          | mz_system

SELECT mz_materialized_views.name AS view_name, mz_roles.name AS role_name
  FROM mz_materialized_views
  LEFT JOIN mz_roles ON mz_materialized_views.owner_id = mz_roles.id
  WHERE mz_materialized_views.name = 'v';

 view_name | role_name 
-----------+-----------
 v         | mz_system


ALTER TABLE t OWNER TO joe;

SELECT mz_tables.name AS table_name, mz_roles.name AS role_name
  FROM mz_tables
  LEFT JOIN mz_roles ON mz_tables.owner_id = mz_roles.id
  WHERE mz_tables.name = 't';

 table_name | role_name 
------------+-----------
 t          | joe


SELECT mz_materialized_views.name AS view_name, mz_roles.name AS role_name
  FROM mz_materialized_views
  LEFT JOIN mz_roles ON mz_materialized_views.owner_id = mz_roles.id
  WHERE mz_materialized_views.name = 'v';

 view_name | role_name 
-----------+-----------
 v         | mz_system

```

`v` depends on `t`, but changing `t`'s owner doesn't affect `v`. Similarly, subsources depend on sources, but changing the source's owner doesn't affect the subsource.

@jkosh44 Can you validate that this behavior is correct?

In testing this, though, I have realized this PR prevents changing ownership of a subsource directly. I'll fix that in an additional commit and push that back up.

Reply:
@jkosh44 ping

Reply:
Sorry for the delay in responding. It could go either way here. Changing the owner of a table does not affect the owner of any dependent views. However, changing the owner of a table does also change the owner of all indexes on that table. Similarly, changing the owner of cluster changes the owner of all replicas within that cluster. This behave usually matches the behavior of `DROP`. `DROP TABLE _` drops the table and all indexes without the need for `CASCADE`. Similarly, `DROP CLUSTER _` drops the cluster and all replicas without the need for `CASCADE`.

So if `DROP SOURCE _` works without `CASCADE` then I think the behavior of this PR is correct. Otherwise, we should revert to the old behavior.

Reply:
The [SQL council concluded](https://materializeinc.slack.com/archives/C063H5S7NKE/p1713344604109359) that dropping a source w/ subsources requiring `CASCADE` is acceptable, so going to leave the PR as is.